### PR TITLE
Update javadocs because EOS is now available

### PR DESCRIPTION
**Pull request by miguno**

### CODE REVIEW

**Comment by miguno** [comment]
Ping @guozhangwang @enothereska @mjsax @dguy @bbejeck

File: kafka-streams/src/test/java/io/confluent/examples/streams/EventDeduplicationLambdaIntegrationTest.java [hunk 1]
```diff
@@ -62,15 +62,12 @@
...
- * events under normal, non-failure conditions; in such cases, an event de-duplication approach as
- * shown in this example is helpful.
- *
- * https://cwiki.apache.org/confluence/display/KAFKA/KIP-98+-+Exactly+Once+Delivery+and+Transactional+Messaging
- * https://cwiki.apache.org/confluence/display/KAFKA/KIP-129%3A+Streams+Exactly-Once+Semantics
+ * IMPORTANT:  Since Kafka 0.11, the Kafka Streams API supports exactly-once processing semantics.
+ * With this feature available, most use cases will no longer need to worry about duplicates or
+ * duplicate processing because, typically, such duplicates only happen in the face of failures.
+ * That said, there will still be some scenarios where the upstream data producers may generate
+ * duplicate events under normal, non-failure conditions; in such cases, an event de-duplication
```
Comment (mjsax) [inline kafka-streams/src/test/java/io/confluent/examples/streams/EventDeduplicationLambdaIntegrationTest.java]: `where the upstream data producers may generate duplicate events under normal, non-failure conditions`

This sound wrong IMHO, as it's indicates that the producer itself does generate duplicates what is not the case. With EOS, duplicates only occur, if the "producer application" does write duplicates.

**Comment by miguno** [comment]
@mjsax Makes sense.  I clarified the text, feel free to review again.

**Comment by miguno** [comment]
Also merged into `3.4.x` and `master`.

**Pull request merged**